### PR TITLE
Update readme for Azure.Monitor.Query

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/22188

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 1]
```diff
@@ -33,8 +33,46 @@ To interact with the Azure Monitor service, create an instance of a [TokenCreden
...
+Here is a heirarchy of the LogsQueryResult response:
+
+```
+LogsQueryResult
+|---Statistics
+|---Visualization
+|---Error
+|---AllTables (list of `LogsTable` objects)
+    |---Name
+    |---InternalRows
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/README.md:45]: This is not public

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 2]
```diff
@@ -33,8 +33,46 @@ To interact with the Azure Monitor service, create an instance of a [TokenCreden
 ## Key concepts
 
 - `LogsQueryClient` - Client that provides methods to query logs from Azure Monitor Logs.
+Here is a heirarchy of the LogsQueryResult response:
+
+```
+LogsQueryResult
+|---Statistics
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/README.md:40]: This is not a property

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 3]
```diff
@@ -33,8 +33,46 @@ To interact with the Azure Monitor service, create an instance of a [TokenCreden
...
+|---Interval
+|---Namespace
+|---ResourceRegion
+|---Metrics (list of `MetricResult` objects)
+    |---Id
+    |---Type
+    |---LocalizedName
+    |---Name
+    |---DisplayDescription
+    |---ErrorCode
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/README.md:67]: We only need to list public properties here.

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 4]
```diff
@@ -33,8 +33,46 @@ To interact with the Azure Monitor service, create an instance of a [TokenCreden
 ## Key concepts
 
 - `LogsQueryClient` - Client that provides methods to query logs from Azure Monitor Logs.
+Here is a heirarchy of the LogsQueryResult response:
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:36]:
```suggestion
Here's a hierarchy of the `LogsQueryResult` response object:
```

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 5]
```diff
@@ -33,8 +33,46 @@ To interact with the Azure Monitor service, create an instance of a [TokenCreden
...
+    |---Name
+    |---InternalRows
+    |---Columns (list of `LogsTableColumn` objects)
+        |---Name
+        |---Type
+```
+
 - `MetricsQueryClient` - Client that provides methods to query metrics from Azure Monitor Metrics.
 
+Here is a heirarchy of the MetricsQueryResult response:
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:53]:
```suggestion
Here's a hierarchy of the `MetricsQueryResult` response object:
```

**Review by scottaddie — CHANGES_REQUESTED** [review]
You'll want to rebase. My README PR merged this morning, and it introduced some big changes in this file.

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 6]
```diff
@@ -307,6 +318,34 @@ foreach (var metric in results.Value.Metrics)
...
 ```
 
+#### Handle metrics query response
+The metrics query API returns a `MetricsQueryResult` object. The `MetricsQueryResult` object contains 
+properties such as a list of `MetricResult`-typed objects, `Cost`, `Namespace`, `ResourceRegion`, `TimeSpan`, and `Interval`. The 
+`MetricResult` objects list can be accessed using the `metrics` param. Each `MetricResult` object in this list
+contains a list of `TimeSeries` objects. Each `TimeSeries` contains `Metadatavalues` and `Values`
+properties. 
+
+Here's a hierarchy of the response:
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:328]:
```suggestion

The metrics query API returns a `MetricsQueryResult` object. The `MetricsQueryResult` object contains properties such as a list of `MetricResult`-typed objects, `Cost`, `Namespace`, `ResourceRegion`, `TimeSpan`, and `Interval`. The `MetricResult` objects list can be accessed using the `metrics` param. Each `MetricResult` object in this list contains a list of `MetricTimeSeriesElement` objects. Each `MetricTimeSeriesElement` object contains `Metadata` and `Values` properties.

Here's a hierarchy of the response:
```

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 7]
```diff
@@ -307,6 +318,34 @@ foreach (var metric in results.Value.Metrics)
...
+|---TimeSpan
+|---Metrics (list of `MetricResult` objects)
+    |---Id
+    |---Type
+    |---Name
+    |---DisplayDescription
+    |---Error
+    |---Unit
+    |---TimeSeries (list of `MetricTimeSeriesElement` objects)
+        |---Metadatavalues
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:345]: `Metadatavalues` is private.

```suggestion
        |---Metadata
```

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 8]
```diff
@@ -38,20 +38,10 @@ For examples of Logs and Metrics queries, see the [Examples](#examples) section.
 
 ## Key concepts
 
-### Logs query rate limits and throttling
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:41]: Keep the "Logs query rate limits and throttling" and "Metrics data structure" sections here. They're present in the other languages' READMEs. For example: https://github.com/Azure/azure-sdk-for-java/blob/6a429425c0a498469a368270d77fa02f673adc9f/sdk/monitor/azure-monitor-query/README.md#L93-L106

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 9]
```diff
@@ -38,20 +38,10 @@ For examples of Logs and Metrics queries, see the [Examples](#examples) section.
...
-
-Each set of metric values is a time series with the following characteristics:
-
-- The time the value was collected
-- The resource associated with the value
-- A namespace that acts like a category for the metric
-- A metric name
-- The value itself
-- Some metrics may have multiple dimensions as described in multi-dimensional metrics. Custom metrics can have up to 10 dimensions.
+- `MetricsQueryClient` - Client that provides methods to query metrics from Azure Monitor Metrics.
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:44]:
```suggestion
```

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 10]
```diff
@@ -38,20 +38,10 @@ For examples of Logs and Metrics queries, see the [Examples](#examples) section.
 
 ## Key concepts
 
-### Logs query rate limits and throttling
+- `LogsQueryClient` - Client that provides methods to query logs from Azure Monitor Logs.
+Here's a heirarchy of the LogsQueryResult response:
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:42]:
```suggestion
```

**Review by scottaddie — CHANGES_REQUESTED** [review]

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 11]
```diff
@@ -101,6 +103,24 @@ foreach (var row in table.Rows)
 }
 ```
 
+#### Handle logs query response
+The `Query` method returns the `LogsQueryResult`, while the `QueryBatch` method returns the `LogsBatchQueryResult`.
+Here's a hierarchy of the response:
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:108]:
```suggestion
#### Handle logs query response

The `Query` method returns the `LogsQueryResult`, while the `QueryBatch` method returns the `LogsBatchQueryResult`. Here's a hierarchy of the response:
```

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 12]
```diff
@@ -307,6 +327,30 @@ foreach (var metric in results.Value.Metrics)
 }
 ```
 
+#### Handle metrics query response
+The metrics query API returns a `MetricsQueryResult` object. The `MetricsQueryResult` object contains properties such as a list of `MetricResult`-typed objects, `Cost`, `Namespace`, `ResourceRegion`, `TimeSpan`, and `Interval`. The `MetricResult` objects list can be accessed using the `metrics` param. Each `MetricResult` object in this list contains a list of `TimeSeries` objects. Each `TimeSeries` contains `Metadatavalues` and `Values` properties.
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:331]:
```suggestion
#### Handle metrics query response

The metrics query API returns a `MetricsQueryResult` object. The `MetricsQueryResult` object contains properties such as a list of `MetricResult`-typed objects, `Cost`, `Namespace`, `ResourceRegion`, `TimeSpan`, and `Interval`. The `MetricResult` objects list can be accessed using the `metrics` param. Each `MetricResult` object in this list contains a list of `MetricTimeSeriesElement` objects. Each `MetricTimeSeriesElement` object contains `Metadata` and `Values` properties. 
```

**Review by scottaddie — APPROVED** [review]
One last round of changes, and then this looks good.

**Review by scottaddie — APPROVED** [review]

**Pull request merged**